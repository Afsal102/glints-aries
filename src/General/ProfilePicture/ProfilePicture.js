/* @flow */

import React from 'react';
import { ProfilePictureContainer, ProfilePictureContent } from '../../Style/General/ProfilePictureStyle';

const ProfilePicture = (props: Props) => {
  const {
    editable,
    children,
    className,
    ...defaultProps
  } = props;

  return (
    <ProfilePictureContainer
      className={className}
      tabIndex="0"
      role="presentation"
      {...defaultProps}
    >
      <ProfilePictureContent
        editable={editable}
        tabIndex="-1"
      >
        {children}
      </ProfilePictureContent>
    </ProfilePictureContainer>
  );
};

type Props = {
  editable: boolean,
  className: string,
  children: string,
};

export default ProfilePicture;
