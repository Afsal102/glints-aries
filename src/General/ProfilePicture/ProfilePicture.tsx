import * as React from 'react';

import classNames from 'classnames';

import { EditIcon } from '../Icon/components';

import {
  ProfilePictureContainer,
  ProfilePictureContent,
} from '../../Style/General/ProfilePictureStyle';

const ProfilePicture = ({
  editable,
  children,
  className,
  ...defaultProps
}: Props) => {
  const [isHover, setIsHover] = React.useState<boolean>(false);

  const handleMouseEnter = () => {
    if (editable) {
      setIsHover(true);
    }
  };

  const handleMouseLeave = () => {
    if (editable) {
      setIsHover(false);
    }
  };

  return (
    <ProfilePictureContainer
      className={classNames('aries-profile', className)}
      role={editable ? 'button' : 'presentation'}
      aria-label="Profile Picture"
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
      tabIndex={editable ? 0 : undefined}
      {...defaultProps}
    >
      <ProfilePictureContent
        editable={editable}
        tabIndex={editable ? -1 : undefined}
      >
        {isHover && <EditIcon color="white" />}
        {children}
      </ProfilePictureContent>
    </ProfilePictureContainer>
  );
};

interface Props
  extends React.ComponentPropsWithoutRef<typeof ProfilePictureContainer> {
  editable?: boolean;
  children: React.ReactNode;
}

export default ProfilePicture;
