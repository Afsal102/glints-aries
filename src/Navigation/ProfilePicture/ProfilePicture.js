/* @flow */

import React from 'react';
import { Container } from '../../Style/ProfilePictureStyle';

const ProfilePicture = (props: Props) => {
  const {
    editable,
    children,
    className,
    ...defaultProps
  } = props;

  return (
    <Container className={className} editable={editable} {...defaultProps}>
      {children}
    </Container>
  );
};

type Props = {
  editable: boolean,
  className: string,
  children: string,
};

export default ProfilePicture;
