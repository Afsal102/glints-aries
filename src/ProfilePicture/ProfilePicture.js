/* @flow */
import React, { Fragment } from 'react';
import { ProfilePictureWrapper, ProfilePictureImage } from '../Style/ProfilePictureStyle';

const ProfilePicture = (props: Props) => {
  const { size, assetUrl } = props;

  return (
    <Fragment>
        <ProfilePictureWrapper size={size}>
            <ProfilePictureImage src={assetUrl} />
        </ProfilePictureWrapper>
    </Fragment>
  );
}

export default ProfilePicture;
