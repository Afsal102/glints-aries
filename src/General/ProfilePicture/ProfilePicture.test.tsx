import * as React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent } from '@testing-library/react';

import ProfilePicture from './ProfilePicture';

describe('<ProfilePicture/> snapshots with prop editable', () => {
  const matchSnapshotWithPropEditable = (editable: boolean) => {
    test(`editable ${editable}`, () => {
      const { asFragment } = render(
        <ProfilePicture editable={editable}>
          <img src="/mock.jpg" />
        </ProfilePicture>
      );
      expect(asFragment()).toMatchSnapshot();
    });
  };

  [true, false].forEach(editable => matchSnapshotWithPropEditable(editable));
});

describe('<ProfilePicture/> snapshots with mouse event', () => {
  test('mouseenter and mouseleave when is editable', () => {
    const { asFragment, queryByRole } = render(
      <ProfilePicture editable={true}>
        <img src="/mock.jpg" />
      </ProfilePicture>
    );

    const container = queryByRole('button');
    fireEvent.mouseEnter(container);
    expect(asFragment()).toMatchSnapshot();
    fireEvent.mouseLeave(container);
    expect(asFragment()).toMatchSnapshot();
  });

  test('mouseenter and mouseleave when is not editable', () => {
    const { asFragment, queryByRole } = render(
      <ProfilePicture editable={false}>
        <img src="/mock.jpg" />
      </ProfilePicture>
    );

    const container = queryByRole('presentation');
    fireEvent.mouseEnter(container);
    expect(asFragment()).toMatchSnapshot();
    fireEvent.mouseLeave(container);
    expect(asFragment()).toMatchSnapshot();
  });
});
