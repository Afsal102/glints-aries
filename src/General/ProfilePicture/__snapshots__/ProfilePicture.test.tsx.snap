// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfilePicture/> snapshots with mouse event mouseenter and mouseleave when is editable 1`] = `
<DocumentFragment>
  .c2 {
  vertical-align: middle;
}

.c1 {
  position: relative;
  z-index: 1;
  width: 5em;
  height: 5em;
  border-radius: 50%;
  outline: none;
  cursor: pointer;
}

.c1 svg {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1:hover:before {
  content: '';
  position: absolute;
  background-color: rgba(0,0,0,.5);
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  border-radius: 50%;
}

.c1:after {
  content: '';
  position: absolute;
  z-index: -1;
  background: url(test-file-stub);
  height: 100%;
  width: 100%;
  border-radius: 50%;
  top: 4px;
  left: 4px;
}

.c1 img {
  object-fit: cover;
  border-radius: 50%;
  width: 100%;
  height: 100%;
  z-index: 1;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
    aria-label="Profile Picture"
    class="c0 aries-profile"
    role="button"
    tabindex="0"
  >
    <div
      class="c1"
      tabindex="-1"
    >
      <svg
        class="c2"
        data-testid="icon-svg"
        fill="white"
        height="1em"
        viewBox="0 0 100 100"
        width="1em"
      >
        <path
          d="M0 79.22l61.299-61.298 20.779 20.78L20.779 100H0V79.22zm98.182-56.623l-10.39 10.39-20.779-20.78 10.39-10.389c2.424-2.424 4.935-2.424 7.532 0l13.247 13.247c2.424 2.597 2.424 5.108 0 7.532z"
        />
      </svg>
      <img
        src="/mock.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ProfilePicture/> snapshots with mouse event mouseenter and mouseleave when is editable 2`] = `
<DocumentFragment>
  .c1 {
  position: relative;
  z-index: 1;
  width: 5em;
  height: 5em;
  border-radius: 50%;
  outline: none;
  cursor: pointer;
}

.c1 svg {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1:hover:before {
  content: '';
  position: absolute;
  background-color: rgba(0,0,0,.5);
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  border-radius: 50%;
}

.c1:after {
  content: '';
  position: absolute;
  z-index: -1;
  background: url(test-file-stub);
  height: 100%;
  width: 100%;
  border-radius: 50%;
  top: 4px;
  left: 4px;
}

.c1 img {
  object-fit: cover;
  border-radius: 50%;
  width: 100%;
  height: 100%;
  z-index: 1;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
    aria-label="Profile Picture"
    class="c0 aries-profile"
    role="button"
    tabindex="0"
  >
    <div
      class="c1"
      tabindex="-1"
    >
      <img
        src="/mock.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ProfilePicture/> snapshots with mouse event mouseenter and mouseleave when is not editable 1`] = `
<DocumentFragment>
  .c1 {
  position: relative;
  z-index: 1;
  width: 5em;
  height: 5em;
  border-radius: 50%;
}

.c1 svg {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1:after {
  content: '';
  position: absolute;
  z-index: -1;
  background: url(test-file-stub);
  height: 100%;
  width: 100%;
  border-radius: 50%;
  top: 4px;
  left: 4px;
}

.c1 img {
  object-fit: cover;
  border-radius: 50%;
  width: 100%;
  height: 100%;
  z-index: 1;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
    aria-label="Profile Picture"
    class="c0 aries-profile"
    role="presentation"
  >
    <div
      class="c1"
    >
      <img
        src="/mock.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ProfilePicture/> snapshots with mouse event mouseenter and mouseleave when is not editable 2`] = `
<DocumentFragment>
  .c1 {
  position: relative;
  z-index: 1;
  width: 5em;
  height: 5em;
  border-radius: 50%;
}

.c1 svg {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1:after {
  content: '';
  position: absolute;
  z-index: -1;
  background: url(test-file-stub);
  height: 100%;
  width: 100%;
  border-radius: 50%;
  top: 4px;
  left: 4px;
}

.c1 img {
  object-fit: cover;
  border-radius: 50%;
  width: 100%;
  height: 100%;
  z-index: 1;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
    aria-label="Profile Picture"
    class="c0 aries-profile"
    role="presentation"
  >
    <div
      class="c1"
    >
      <img
        src="/mock.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ProfilePicture/> snapshots with prop editable editable false 1`] = `
<DocumentFragment>
  .c1 {
  position: relative;
  z-index: 1;
  width: 5em;
  height: 5em;
  border-radius: 50%;
}

.c1 svg {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1:after {
  content: '';
  position: absolute;
  z-index: -1;
  background: url(test-file-stub);
  height: 100%;
  width: 100%;
  border-radius: 50%;
  top: 4px;
  left: 4px;
}

.c1 img {
  object-fit: cover;
  border-radius: 50%;
  width: 100%;
  height: 100%;
  z-index: 1;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
    aria-label="Profile Picture"
    class="c0 aries-profile"
    role="presentation"
  >
    <div
      class="c1"
    >
      <img
        src="/mock.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ProfilePicture/> snapshots with prop editable editable true 1`] = `
<DocumentFragment>
  .c1 {
  position: relative;
  z-index: 1;
  width: 5em;
  height: 5em;
  border-radius: 50%;
  outline: none;
  cursor: pointer;
}

.c1 svg {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1:hover:before {
  content: '';
  position: absolute;
  background-color: rgba(0,0,0,.5);
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  border-radius: 50%;
}

.c1:after {
  content: '';
  position: absolute;
  z-index: -1;
  background: url(test-file-stub);
  height: 100%;
  width: 100%;
  border-radius: 50%;
  top: 4px;
  left: 4px;
}

.c1 img {
  object-fit: cover;
  border-radius: 50%;
  width: 100%;
  height: 100%;
  z-index: 1;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
    aria-label="Profile Picture"
    class="c0 aries-profile"
    role="button"
    tabindex="0"
  >
    <div
      class="c1"
      tabindex="-1"
    >
      <img
        src="/mock.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;
